tarorpc/assetwalletrpc/assetwallet.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+
+import "taro.proto";
+
+package assetwalletrpc;
+
+option go_package = "github.com/lightninglabs/tarorpc/assetwalletrpc";
+
+service AssetWallet {
+    /*
+    FundVirtualPsbt selects inputs from the available asset commitments to fund
+    a virtual transaction matching the template.
+    */
+    rpc FundVirtualPsbt (FundVirtualPsbtRequest)
+        returns (FundVirtualPsbtResponse);
+
+    /*
+    SignVirtualPsbt signs the inputs of a virtual transaction and prepares the
+    commitments of the inputs and outputs.
+    */
+    rpc SignVirtualPsbt (SignVirtualPsbtRequest)
+        returns (SignVirtualPsbtResponse);
+
+    /*
+    AnchorVirtualPsbts merges and then commits multiple virtual transactions in
+    a single BTC level anchor transaction.
+
+    TODO(guggero): Actually implement accepting and merging multiple
+    transactions.
+    */
+    rpc AnchorVirtualPsbts (AnchorVirtualPsbtsRequest)
+        returns (tarorpc.SendAssetResponse);
+}
+
+message FundVirtualPsbtRequest {
+    oneof template {
+        /*
+        Use an existing PSBT packet as the template for the funded PSBT.
+
+        TODO(guggero): Actually implement this. We can't use the "reserved"
```

Comment:
Missing link here is parsing? Can def push that out to other PRs since this series itself is already somewhat hefty.

Reply:
Not just parsing, but a whole bunch of additional validations (e.g. only one input, first output set as change output, script keys set properly, coin selected properly and so on). So yeah, going to leave it for now.

Reply:
We'll start to support this to enable the interactive send case. But I decided to split that into a separate PR since more changes are necessary than initially expected.